.gitpod.yml

```diff
@@ -0,0 +1,42 @@
+# Commands to start on workspace startup
+
+ports:
+  # when port 8000 becomes available open a browser preview
+  # of the running django pattern library instance
+  - port: 8000
+    onOpen: open-preview
+
+tasks:
+  # initialise and install the python part of the app by installing poetry
+  # and the python dependencies
+  - init: |
+      clear
+      pip install poetry
+      poetry install
+      gp sync-done python
+    # Start the server for testing
+    command: poetry run django-admin runserver --settings=tests.settings.dev --pythonpath=.
+    name: Runserver
+
+  # install the node part of the app by installing the required
+  # node version and npm dependencies
+  - init: |
+      clear
+      nvm install
+      npm install
+    # Build the assets
+    command: npm run build
+    name: Node
+    openMode: split-right
+
+  # when the python part of the app is ready...
+  - init: |
+      clear
+      echo "Waiting for python..."
+      gp sync-await python
+    # run the tests
+    command: |
+      poetry run ./runtests.py
+      poetry run django-admin render_patterns --settings=tests.settings.dev --pythonpath=.
+    name: Tests
+    openMode: split-right
```

Comment:
I don’t totally get this. Is this starting the tests automatically upon starting the development environment in Gitpod? I’ve not seen this done in other projects, is there a reason why we would want this in Gitpod?

Reply:
Again, a personal preference. But it's nice to know that before you start any work the tests ran OK and it's a reminder that there are tests in place that you should run.

A nice side effect is you get the coverage report saved at the start.

Happy to remove this though if you feel it's not required.

Reply:
Cool! I’d be interested to see this in practice.